editor/src/components/editor/store/dispatch.tsx

```diff
@@ -770,24 +770,20 @@ function findMatchingTemplatePath(
       newComponents.elements,
       pathToUpdate as InstancePath,
     ) != null
-  if (pathStillExists || scenePathStillExists) {
+  if (pathStillExists || (TP.isScenePath(pathToUpdate) && scenePathStillExists)) {
     return pathToUpdate
   }
 
   const parentPath = TP.parentPath(pathToUpdate)
-  const oldParent = MetadataUtils.getElementByInstancePathMaybe(
-    oldComponents.elements,
-    parentPath as InstancePath,
-  )
-  const newParent = MetadataUtils.getElementByInstancePathMaybe(
-    newComponents.elements,
-    parentPath as InstancePath,
-  )
-  if (oldParent != null && newParent != null) {
-    const oldChildIndex = oldParent.children.findIndex((p) => TP.pathsEqual(p, pathToUpdate))
-    const potentialNewPath = newParent.children[oldChildIndex]
-    return potentialNewPath ?? null
+  if (parentPath == null) {
+    return null
```

Comment:
maybe it's worth leaving a comment here that this means we will not even attempt to salvage selection for root elements

```suggestion
    return null // TODO this means we don't attempt to salvage the selection of root elements
```

Reply:
Good catch! I've updated the logic now to handle root paths